internal/bootstrap/deployer.go

```diff
@@ -454,9 +483,16 @@ func (b *baseDeployer) AddControllerApplication(ctx context.Context, info Deploy
 	stateOrigin.Hash = ""
 	stateOrigin.ID = ""
 
-	_, err = b.charmUploader.PrepareCharmUpload(info.URL.String())
-	if err != nil {
-		return nil, errors.Trace(err)
+	if info.URL.Schema == charm.Local.String() {
+		_, err = b.charmUploader.PrepareLocalCharmUpload(info.URL.String())
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+	} else {
+		_, err = b.charmUploader.PrepareCharmUpload(info.URL.String())
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
 	}
```

Comment:
This is to keep the controller from crashing out. It will be removed pretty soon.